packages/data-store/src/typeorm/entities/document.entity.ts

```diff
@@ -0,0 +1,121 @@
+import {
+  Entity,
+  Column,
+  PrimaryGeneratedColumn,
+  PrimaryColumn,
+  ManyToMany,
+  JoinTable,
+  Relation,
+} from 'typeorm';
+import assert from 'assert';
+import {getDataSource} from '../index';
+import {DocumentTypeEntity} from './document-type.entity';
+import {logger} from '../../logger';
+import {getActiveWallet} from '../../configs';
+
+@Entity()
+export class DocumentEntity {
+  @PrimaryColumn('text')
+  id: string;
+
+  @Column('text', {
+    nullable: true,
+  })
+  walletId: string;
+
+  // TODO: check if that's required, information already available on data
+  @Column('simple-array')
+  type: string[];
+
+  @ManyToMany(() => DocumentTypeEntity, type => type?.documents, {
+    cascade: ['insert'],
+    eager: true,
+    nullable: true,
+  })
+  @JoinTable()
+  _typeRel: Relation<DocumentTypeEntity[]>;
+
+  @Column('blob')
+  data: string;
+}
+
+export async function getOrCreateDocumentTypes(
+  types,
+): Promise<DocumentTypeEntity[]> {
+  logger.debug(`getOrCreateDocumentTypes: ${JSON.stringify(types)}`);
+
+  if (!Array.isArray(types)) {
+    types = [types];
+  }
+
+  const typeRepository = getDataSource().getRepository(DocumentTypeEntity);
+  const typeEntityList = [];
+  for (const type of types) {
+    let typeEntity = await typeRepository.findOne({
+      where: {
+        id: type,
+      },
+    });
+
+    if (!typeEntity) {
+      typeEntity = typeRepository.create({
+        id: type,
+      });
+
+      await typeRepository.save(typeEntity);
+    }
+
+    typeEntityList.push(typeEntity);
+  }
+
+  return typeEntityList;
+}
+
+export async function createDocument(json: any): Promise<DocumentEntity> {
+  assert(!!json.id, 'Document must have an id');
+
+  // TODO: get walletId from context
+  const walletId = await getActiveWallet();
+
```

Comment:
when are these TODOs getting handled?

Reply:
Actually this looks correct, just forgot to remove the TODO comment. The wallet id was hardcoded when I added the TODO